### PR TITLE
style: Fix if-else-block-instead-of-if-exp (SIM108) (part 3)

### PR DESCRIPTION
**Pull request by echoix**
Last PR of this series. It finishes by enabling checking the rule with ruff.

PRs in this series: #4562, #4561

Ruff rule: https://docs.astral.sh/ruff/rules/if-else-block-instead-of-if-exp

### CODE REVIEW

**Comment by echoix** [comment]
On my fork I think it's bandit that will complain about the use of eval, that is only moved around

File: gui/wxpython/dbmgr/base.py [hunk 1]
```diff
@@ -1542,10 +1542,7 @@ def OnDataItemEdit(self, event):
                     column = tlist.columns[columnName[i]]
                     if len(values[i]) > 0:
                         try:
-                            if missingKey is True:
-                                idx = i - 1
-                            else:
-                                idx = i
+                            idx = i - 1 if missingKey is True else i
```
Comment (petrasovaa) [inline gui/wxpython/dbmgr/base.py:1545]:
```suggestion
                            idx = i - 1 if missingKey else i
```

File: gui/wxpython/nviz/mapwindow.py [hunk 2]
```diff
@@ -382,10 +382,7 @@ def OnEraseBackground(self, event):
 
     def OnSize(self, event):
         size = self.GetClientSize()
-        if CheckWxVersion(version=[2, 9]):
```
Comment (petrasovaa) [inline gui/wxpython/nviz/mapwindow.py:385]: We may as well just remove the check, it's too old version

File: gui/wxpython/core/treemodel.py [hunk 3]
```diff
@@ -294,15 +294,12 @@ def __init__(self, label=None, data=None):
     def label(self):
         return self._label
 
-    def match(self, key, value, case_sensitive=False):
+    def match(self, key, value, case_sensitive=False) -> bool:
```
Comment (petrasovaa) [inline gui/wxpython/core/treemodel.py:297]: Do we need `from __future__ import annotations` here?

File: python/grass/script/core.py
Comment (petrasovaa) [inline python/grass/script/core.py:1]: Why this change in this PR?

Comment (echoix) [reply python/grass/script/core.py:1]: I explained in #4561, in order to have more precise validations on what simplifications could be done in regards to the types. I analyzed these and greatly helped, as it allowed type checking to understand that some variables are strings (coming from the options), and that the values from the flags are bools, not any type. So some of the conditions could assume the bool type and just use "or" instead.

File: gui/wxpython/core/treemodel.py (hunk 3 above)
Comment (echoix) [reply gui/wxpython/core/treemodel.py:297]: No, the future is used when we want to use `|` instead of `Optional[]` (that needs to be imported too), plus more other syntax that was supposed to be mandatory in 3.10, but still postponed. (That future exists since 3.7).

https://docs.python.org/3/library/__future__.html#module-contents

It also uses the new behavior of considering the types as strings without evaluating for better performance, instead of evaluating the types. Type checking doesn't isn't enforced at runtime anyways, so post-poneing makes sense.

In this file, bool doesn't need extra workarounds to work correctly.

Type annotations isn't new, it was there in Python 3.5, but supported syntax evolved to be more useful

File: gui/wxpython/nviz/mapwindow.py (hunk 2 above)
Comment (echoix) [reply gui/wxpython/nviz/mapwindow.py:385]: Without docs, I'm not sure what does the check return true for (to choose what branch). If you don't know I'll read later

File: gui/wxpython/dbmgr/base.py (hunk 1 above)
Comment (echoix) [reply gui/wxpython/dbmgr/base.py:1545]: I see in the scope of the function that missingKey is either True or False, so it should be fine with that suggestion

File: gui/wxpython/nviz/mapwindow.py [hunk 4]
```diff
@@ -382,10 +382,7 @@ def OnEraseBackground(self, event):
...
-            context = self.context
-        else:
-            context = self.GetContext()
+        context = self.context if CheckWxVersion(version=[2, 9]) else self.GetContext()
```
Comment (petrasovaa) [inline gui/wxpython/nviz/mapwindow.py:385]: _True if current wx version is greater or equal than specified version otherwise False_

```suggestion
        context = self.context
```
But the check is in several places, so perhaps there is no point in removing it here only.

**Review by petrasovaa — APPROVED** [review]

File: gui/wxpython/nviz/mapwindow.py (hunk 4 above)
Comment (echoix) [reply gui/wxpython/nviz/mapwindow.py:385]: Ok, so let me keep it for the follow up PR(s) that I plan to do that builds upon this series of 3 PRs. (Problems we found in code, but not related to the changes specifically).

I plan to work on it somewhere this weekend, so I hope to have the open PRs merged to be able to advance without having to redo the work because of conflicts
